assets/js/modules/analytics-4/components/setup/SetupForm.js

```diff
@@ -69,14 +70,22 @@ export default function SetupForm( { finishSetup } ) {
 	const viewContext = useViewContext();
 	const setupFlowRefreshEnabled = useFeature( 'setupFlowRefresh' );
 
+	const showProgress = getQueryArg( location.href, 'showProgress' );
+
 	const keyMetricsSetupURL = useSelect( ( select ) => {
 		if ( ! setupFlowRefreshEnabled ) {
 			return undefined;
 		}
 
-		return select( CORE_SITE ).getAdminURL(
+		const url = select( CORE_SITE ).getAdminURL(
 			'googlesitekit-key-metrics-setup'
 		);
+
+		return showProgress
```

Comment:
Although technically out of scope, we've missed passing the `showProgress` query parameter to the Key Metrics setup screen, so to be pragmatic I've pushed a commit to include it here.